brainscore/benchmarks/moeller2017.py

```diff
@@ -0,0 +1,503 @@
+        '''
+        Sample one voxel outside of face patch
+        1. make a list of voxels where neither the voxel nor its close neighbors are in a face patch
+        2. randomly sample from list
+        :param selectivity_assembly:
+        :param radius: determining the neighborhood size in mm of each voxel that cannot be selective
+        :return: x, y location of voxel outside FP
+        '''
+        not_selective_voxels = selectivity_assembly[selectivity_assembly.values < DPRIME_THRESHOLD_SELECTIVITY]
+        voxels = []
+        for voxel in not_selective_voxels:
+            inside_radius = np.where(np.sqrt(np.square(not_selective_voxels.voxel_x.values - voxel.voxel_x.values) +
+                                             np.square(not_selective_voxels.voxel_y.values - voxel.voxel_y.values))
+                                     < radius)[0]
+            if np.all(not_selective_voxels[inside_radius].values < DPRIME_THRESHOLD_FACE_PATCH):
+                voxels.append(voxel)
+
+        rng = np.random.default_rng(seed=self._seed)
+        voxel = rng.choice(voxels)
+        x, y = voxel.voxel_x, voxel.voxel_y
+        return x, y
+
+    def _collect_target_assembly(self):
+        '''
+        TODO make sure assembly.stimulus_set corresponds to target_stimulus_class
+        TODO stimulus set FP
+        TODO stimulus set training decoder
+        TODO full list of objects
+        TODO make sure assembly has source attribute --> ceiling??
+        Load Data from path + subselect as specified by Experiment
+
+        :return: DataAssembly
+        '''
+        STIMULUS_CLASS_DICT = {
+            'faces': ['faces'],  # 32; 6 expression each
+            'objects': [],  # TODO 28; 3 viewing angles each
+            'non_face_objects_eliciting_FP_response_plus_faces': ['apples_bw', 'citrus_fruits_bw', 'teapots_bw',
+                                                                  'clocks_bw', 'faces_bw'],  # 15; 3/cats
+            'abstract_faces': ['face_cartoons', 'face_linedrawings', 'face_mooneys', 'face_silhouettes'],  # 16; 4/cat
+            'abstract_houses': ['house_cartoons', 'house_linedrawings', 'house_silhouettes', 'face_mooneys',
+                                'face_mooneys_inverted']  # 20; 4/cat
+        }
+
+        assembly = self._load_assembly()
+
+        target_stimulus_class = STIMULUS_CLASS_DICT[self._stimulus_class]
+        target_assembly = assembly.multisel(object_name=target_stimulus_class)  # TODO
+        # TODO add object_ID & image_ID coords; image_ID = expression or viewing angle
+        return target_assembly
+
+    @staticmethod
+    def _load_assembly():
+        '''
+        TODO Load DataArray from path
+        TODO package all data
+        TODO get stimulus set
+        make sure it has a monkey dimension for ceiling computation
+        :return: DataArray
+        '''
+        assembly = None  # TODO
+        return assembly
+
+
+def Moeller2017Exp1():
+    '''
+    EXP1:

```

Comment:
I guess they call it "Exp1", but at least in the comment you can write it out -- I promise we can buy enough storage space for those additional 7 characters